src/hotspot/share/runtime/objectMonitor.cpp

```diff
@@ -2044,6 +2051,13 @@ void ObjectMonitor::notifyAll(TRAPS) {
     return;
   }
 
+  quick_notifyAll(current);
```

Comment:
When the regular versions end up also using the quick versions, a distinction has been lost in translation - is the only difference that the regular versions enter _thread_in_vm, whereas the quick versions enter _thread_in_java? If there is no real difference, consider eliminating the quick versions and referring directly to the regular versions. 

From a JFR perspective, we need threads in state VM or Java for writing events that refer to artifacts; both states are equivalent for epoch management. You can also have threads in native writing JFR events, but only if the event fields are scalars, not foreign keys. However, the thread cannot capture a stack trace.

Reply:
A JavaThread in state VM or Java running the JFR commit code will not to a transition or block for a safepoint. It is not waitfree (yet) however because of the JfrStacktrace_lock.

Reply:
Are we keeping the NoSafepointVerifier in the quick entries?

Reply:
> Are we keeping the NoSafepointVerifier in the quick entries?

Yes we do.